### PR TITLE
Refactoring • re-organize JSDoc parsing files

### PR DESCRIPTION
**Pull request by michaelrambeau**
## Goal

Currently the logic about the parsing of JSDoc comments to extract component meta data is split into a lot of different files.
Files often contain a single function with a single instruction.
As a result we keep jumping between short files in the IDE to follow the logic.

This PR is an attempt to improve a bit the situation by gathering together pieces that should fit together, while keeping in mind the separation of concerns and avoiding creating cluttered big files.

### Before

```
./comments
├── getNodeJsDocTag.ts
├── getNodeJsDocTags.ts
├── getUXPinNamespaceComment.ts
├── getUXpinDocUrlComment.ts
└── hasUXPinComponentComment.ts
```

### After

To sum up the changes, I created a file `jsdoc-helpers` with the low level functions that parse the JSDoc doc comments.

Then I created a file for each tag we parse:
 
- `jsdoc-component.ts` for `@uxpincomponent` tag
- `jsdoc-docurl.ts` for `@uxpindocurl` tag
- `jsdoc-namespace` for `@uxpinnamespace` tag

```
./comments
├── jsdoc-component.ts
├── jsdoc-docurl.ts
├── jsdoc-helpers.ts
└── jsdoc-namespace.ts
```

## How to check

Nothing to check as it's only about moving functions and renaming files, we can rely on the test coverage.

### CODE REVIEW

**Review by Evomatic — APPROVED** [review]
This appears a lot cleaner than what we had before. I remember when i originally worked on, and wondered why we had so many small methods in a single file! LGTM 🧑‍💻
